src/core.rs

```diff
@@ -258,79 +227,92 @@ impl Internal {
-                            new_size = Some(logical_size);
-                        }
-                        _ => {},
-                    },
-                    _ => {},
-                }
-            });
+            let mut new_mouse_pos: Option<PhysicalPosition<f64>> = None;
+
+            // Copy the current states into the previous state for input
+            for (_, val) in &mut input.keys {
+                val.0 = val.1;
+            }
+
+            for (_, val) in &mut input.mouse {
+                val.0 = val.1;
+            }
+
+            match event {
+                Event::WindowEvent { event, .. } => match event {
+                    WindowEvent::CloseRequested => *flow = ControlFlow::Exit,
+                    WindowEvent::KeyboardInput {
+                        input: KeyboardInput {
+                            virtual_keycode: Some(vk),
+                            state,
+                            ..
+                        },
+                        ..
+                    } => {
+                        let key = input.keys.entry(vk)
+                            .or_insert((false, false));
+                        key.1 = state == ElementState::Pressed;
+                    }
+                    WindowEvent::CursorMoved { position, .. } => {
+                        new_mouse_pos = Some(position);
+                    }
+                    WindowEvent::MouseInput { state, button, .. } => {
+                        let button = input.mouse.entry(button)
+                            .or_insert((false, false));
+                        button.1 = state == ElementState::Pressed;
+                    }
+                    WindowEvent::ModifiersChanged(modifiers) => {
+                        input.modifiers = modifiers;
+                    }
+                    WindowEvent::Resized(logical_size) => {
+                        new_size = Some(logical_size);
+                    }
+                    _ => {}
+                },
+                _ => {}
+            }
+
             if let Some(size) = new_size {
-                let dpi_factor = self.gl_window.get_hidpi_factor();
-                let (x, y) = size.to_physical(dpi_factor).into();
-                self.resize_viewport(x, y);
+                self.resize_viewport(size.width, size.height);
                 input.resized = false;
             }
+
             if let Some(pos) = new_mouse_pos {
-                let dpi_factor = self.gl_window.get_hidpi_factor();
-                let (x, y): (f64, f64) = pos.to_physical(dpi_factor).into();
+                let (x, y): (f64, f64) = pos.into();
                 let x_scale = self.fb.buffer_width as f64 / (self.fb.vp_width as f64);
                 let y_scale = self.fb.buffer_height as f64 / (self.fb.vp_height as f64);
                 let mouse_pos = (
                     x * x_scale,
                     // use the OpenGL texture coordinate system instead of window coordinates

```

Comment:
The "OpenGL texture coordinate system" you refer to is the one where the origin is at the bottom left. You inverted the mouse Y so that programs that want the mouse position will get the one that's over the pixel at the specified coordinates - one with the origin at the bottom left.

Reply:
I think invert_y is good for now. What I meant was maybe take `origin: Origin` in config, which would let you put the origin in any corner or the center, or something like that.

Reply:
That would be even more cursed. I think this is fine. The only reason inverted Y exists is because existing applications depend on it... then again, we could just remove it altogether and force everyone to move, as I have with all my breaking changes.